src/StrawberryfieldFilePersisterService.php

```diff
@@ -177,66 +214,72 @@ public function getDestinationUri(
     string $relativefolder
   ) {
 
-    // Default $relativefolder is a 3 char hash generated by a checksum algorith.
-    $current_uri = $file->getFileUri();
-    $uuid = $file->uuid();
+    if ($file && $file->isTemporary()) {
```

Comment:
THIS IS A BUG FIX @giancarlobi  (i need to make an issue so i can tag it)
If i remove the as:structure from an existing ADO's JSON and i save, the system will try to generate a new valid name for the file and will append an extra filetype. Basically: if the file is named
- giacarlo.jpeg
On save for the first time it will be named
-image-giancarlo-theuuid.jpeg
Next time, if i remove the as:structure
it will be named
-image-image-giancarlo-theuuid.jpeg but will still be left in the same place (instead of moved) because of this check
https://github.com/esmero/strawberryfield/pull/89/files#diff-376f27afff874dbd089c366d26b52febR568
Where we don't re-save files that are already persisted. Because, e.g another ADO could be using it, etc.
So with this fix, now, if the file was already persisted means someone (ourselves) already did our jobs and no need to renamed it. I can make this smarter of course!